tuxedo-core/aggregator/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Ident, ItemEnum};
+
+#[proc_macro_attribute]
+pub fn aggregate(attrs: TokenStream, body: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(body as ItemEnum);
+    let original_code = ast.clone();
+
+    // Uncomment this to inspect the ast of the original code.
+    // eprintln!("{:#?}", ast);
+
+    let outer_type = ast.ident;
+    let variant_type_pairs = ast.variants.iter().map(|variant| {
+        // Make sure there is only a single field, and if not, give a helpful error
+        assert!(
+            variant.fields.len() == 1,
+            "Each variant must have a single unnamed field"
+        );
+        (
+            variant.ident.clone(),
+            variant
+                .fields
+                .iter()
+                .next()
+                .expect("exactly one field per variant")
+                .ty
+                .clone(),
+        )
+    });
+    let variants = variant_type_pairs.clone().map(|(v, _t)| v);
+    let inner_types = variant_type_pairs.map(|(_v, t)| t);
+
+    // The macro supports automatically implementing Tuxedo-related traits.
+    // Currently two traits (Verifier and ConstraintChecker) are supported,
+    // as well as no trait auto-implementation. More traits may be supported in the future.
+    let mut tuxedo_trait_impl = quote! {};
+
+    if !attrs.is_empty() {
```

Comment:
Think would be best to separate out each of verifier and constraint checker into their own macros to avoid this growing very very large

Reply:
Agreed. I've done it now.